packages/atomic/src/components/search/result-template-components/atomic-result-timespan/atomic-result-timespan.tsx

```diff
@@ -0,0 +1,71 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Result, ResultTemplatesHelpers} from '@coveo/headless';
+import {Component, Prop, State} from '@stencil/core';
+import dayjs from 'dayjs';
+import {DurationUnitType} from 'dayjs/plugin/duration';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import {ResultContext} from '../result-template-decorators';
+import duration from 'dayjs/plugin/duration';
+dayjs.extend(duration);
+
+/**
+ * The `atomic-result-timespan` component renders the duration of a number result field.
+ */
+@Component({
+  tag: 'atomic-result-timespan',
+})
+export class AtomicResultTimespan {
+  @InitializeBindings()
+  public bindings!: Bindings;
+
+  @ResultContext() private result!: Result;
+
+  @State() public error!: Error;
+
+  /**
+   * The result field which the component should use.
+   * This will look for the field in the Result object first, and then in the Result.raw object.
+   * It is important to include the necessary field in the ResultList component.
+   */
+  @Prop({reflect: true}) field!: string;
+  /**
+   * The unit of the field value.
+   * Available units: https://day.js.org/docs/en/durations/creating
+   */
+  @Prop({reflect: true}) unit = 'ms';
+  /**
+   * Available formats: https://day.js.org/docs/en/durations/format
+   */
+  @Prop({reflect: true}) format = 'HH:mm:ss';
```

Comment:
I think it would be more useful if we could default to some kind of automatic mode, similar to what you'd find above GitHub comments. As in, it would switch between "seconds", "minutes", "hours", "days", "months", etc based on what's the lowest denominator.

Reply:
I'll see what I can come up with